src/os/os_hardlink_test.go

```diff
@@ -0,0 +1,55 @@
+//go:build !windows && !baremetal && !js && !wasi && !wasip1
```

Comment:
Do we also need to have `!wasm_unknown` in this list?

Reply:
this sounds like a reasonable thing to do